pkg/types/builder.go

```diff
@@ -61,13 +60,12 @@ func (g *Builder) buildResource(res *schema.Resource, names ...string) (*types.N
 	var obsTags []string
 	for _, snakeFieldName := range keys {
 		sch := res.Schema[snakeFieldName]
-		fieldName := strcase.ToCamel(snakeFieldName)
-		lowerCamelFieldName := strcase.ToLowerCamel(snakeFieldName)
-		fieldType, err := g.buildSchema(sch, append(names, fieldName))
+		fieldName := NewNameFromSnake(snakeFieldName)
+		fieldType, err := g.buildSchema(sch, append(names, fieldName.Camel))
 		if err != nil {
-			return nil, nil, errors.Wrapf(err, "cannot infer type from schema of field %s", fieldName)
+			return nil, nil, errors.Wrapf(err, "cannot infer type from schema of field %s", fieldName.Snake)
```

Comment:
No, we calculate the type name at the end of this function only if all succeeds and it doesn't always match a familiar name you derive from Terraform schema because of the uniqueness problem there. And if an error is returned here, we don't print the type hence users would see an error with a type name that doesn't exist because Terraform schema doesn't name types; they are all `map`s with certain keys. So, the only information that is most useful for them at this point is the snake name of the field that they can use to debug the TF schema.